cpp/tests/core/Tensor.cpp

```diff
@@ -106,84 +106,94 @@ TEST_P(TensorPermuteDevices, WithInitList) {
 
     core::Tensor t;
 
-    // 0-D tesnor with given value.
```

Comment:
Add extra test here to test `Scalar`, e.g. mixing different datatypes:
`core::Tensor({{1, 2.2}, {false, 3}}, Dtype::Float32)`